### ISSUE TITLE
Option to wait (blocking) for events as well as polling

### ISSUE THREAD

**Issue by mcavanagh** [Ultz/Silk.NET#165]
# Summary of feature
This is more of a question than a real feature request, as it might just not be what Silk.NET is intended for.  I'm creating some editor software where a significant amount of the time, there's no animation going on.  To avoid unnecessary rendering, I'm only calling `IView.DoRender()` when there's animations still in progress.  However, without the throttling that `SwapBuffers` provides, this now means I'm just calling `IView.DoEvents()` thousands of times per second, which seems equally unnecessary.

Ideally I'd like to be able to switch between polling for events and waiting on them.  `Glfw.WaitEvents()` exists, but isn't exposed via `GlfwWindow`; it only uses `Glfw.PollEvents()`

Is it feasible to also expose `WaitEvents()`, or with Silk.Net primarily being used for games / visually driven apps, is this event-driven style just out of scope?

# Does this have a proposal?
No

**Comment by Perksey** [comment]
This is definitely something we could look to incorporate. Silk.NET, currently, only has a concept of a "game loop" which isn't necessarily geared towards applications (perhaps @Pyrdacor would be interested in this too?)

The concept is pretty baked into the windowing system, but hopefully we can figure something out. I'll add it to this Friday's working group meeting agenda to discuss it.

**Comment by Perksey** [comment]
Good news! I decided to tackle this as part of my windowing improvements PR #175, which means that it'll be a part of tomorrow's update!
